### DISCUSSION TITLE
[5.x]: Allow translation method to be set on the field instance

### DISCUSSION THREAD

**Discussion by johndwells** [craftcms/cms#14607] [Ideas]
Craft 5's field instances are an incredible leap forward in content modeling, this will allow us to greatly reduce the number of custom fields we have to configure!

Right now a field instance can override the field's Label, handle and instructions (as well as conditions). These all make sense because the _context_ of the field's instance, i.e. the entry type it lives in, matters.

We are wondering if the same override could be extended to a field's translation method?  The translation needs of a field are, at least in our instances, also always tied to the entry type it lives in.

🤞

**Comment by brandonkelly** [comment]
On the surface this seems fine, but it’s a little problematic since each field type gets to choose which translation methods it actually supports (including none, like for Matrix and relational fields). So it would be a little confusing if you set the translation method override in a field layout, then change the field’s type to one that doesn’t support the same translation method. At least that’s clear from the field’s edit screen, but you wouldn’t have any visibility into how all the field layouts are being affected.

(Not a deal breaker – there’s a similar bit of awkwardness when switching from a multi-instance-supporting field type to a single-instance one. Just worth considering it.)

**Comment by foxtrotcz** [comment]
I just stumbled upon this limitation.

For example I have very universal Table field called `Table 2Col`. Which is just table with 2 columns and everything else about it depends on context (instance) its used in.
So sometimes the input data can be something which doesnt need translating like numbers with units: 15mm, 12kg, etc.
And sometimes the data are words than need translating.

For now I will just create separate field like `Table 2Col Translatable` and use that so I guess not big deal but I just wanted to say I also noticed this.

Unrelated:
But this brings me to another problem. I cant actually do this. I have content in this field instances that dont support translation and i cant move it another field that would support translation. This issue seems common.
https://github.com/craftcms/cms/discussions/17893
https://github.com/craftcms/cms/discussions/14566
https://github.com/craftcms/cms/discussions/18024
